src/goto-instrument/contracts/contracts.cpp

```diff
@@ -252,13 +252,30 @@ void code_contractst::check_apply_loop_contracts(
     loop_head,
     add_pragma_disable_assigns_check(generated_code));
 
-  // havoc the variables that may be modified
-  assignst assigns;
+  assignst to_havoc;
```

Comment:
Random place for comment: could the tests that were updated in the second commit please be merged into this commit? (That is to say: please fully merge the second commit into this one.)

Reply:
fixed